pkg/event-manager/drivers/entity_handler.go

```diff
@@ -100,13 +106,142 @@ func (h *EntityHandler) Delete(obj entitystore.Entity) error {
 func (h *EntityHandler) Sync(organizationID string, resyncPeriod time.Duration) ([]entitystore.Entity, error) {
 	defer trace.Trace("")()
 
-	return controller.DefaultSync(h.store, h.Type(), organizationID, resyncPeriod, nil)
+	// list entity filter
+	filter := entitystore.FilterEverything().Add(
+		entitystore.FilterStat{
+			Scope:   entitystore.FilterScopeField,
+			Subject: "ModifiedTime",
+			Verb:    entitystore.FilterVerbBefore,
+			Object:  time.Now().Add(-resyncPeriod),
+		},
+		entitystore.FilterStat{
+			Scope:   entitystore.FilterScopeField,
+			Subject: "Status",
+			Verb:    entitystore.FilterVerbIn,
+			Object: []entitystore.Status{
+				entitystore.StatusCREATING, entitystore.StatusUPDATING, entitystore.StatusDELETING,
+				entitystore.StatusERROR,
+				entitystore.StatusREADY,
+			},
+		})
+	syncingEntities, err := controller.DefaultSync(h.store, h.Type(), organizationID, resyncPeriod, filter)
+	if err != nil {
+		return nil, err
+	}
+
+	// sync with k8s, pending entities might change status during sync
+	k := h.backend.(*k8sBackend)
+	var pendingEntities []entitystore.Entity
+	for _, e := range syncingEntities {
+		driver := e.(*entities.Driver)
+
+		switch e.GetStatus() {
+		case entitystore.StatusREADY:
+			// 'real' ready entity will not be processed
+			var k8serr error
+			fullname := getDriverFullName(driver)
+			deployment, err := k.clientset.AppsV1beta1().Deployments(k.config.DriverNamespace).Get(fullname, v1.GetOptions{})
+
+			if err != nil {
+				if k8serrors.IsNotFound(err) {
+					k8serr = &EventdriverErrorDeploymentNotFound{
+						Err: ewrapper.Wrapf(err, "k8s: deployment=%s not found", fullname),
+					}
+				} else {
+					k8serr = &EventdriverErrorUnknown{
+						Err: ewrapper.Wrapf(err, "k8s: deployment=%s unexpected error", fullname),
+					}
+				}
+			} else if deployment.Status.AvailableReplicas == 0 {
+				k8serr = &EventdriverErrorDeploymentNotAvaialble{
```

Comment:
Currently, we set READY state right after deployment is accepted by kubernetes, which means there will be a period of time when available replicas will be 0. This may lead to a weird READY -> ERROR -> READY state flapping, which may be misleading to the user. I guess the solution to that is to ensure there is a replica available before switching from CREATING to READY.